### DISCUSSION TITLE
Delete Category

### DISCUSSION THREAD

**Discussion by gusperandio** [balta-io/forum#343] [Tira Dúvidas] [answered]
### Curso 2806 - Acesso à dados com .NET, C#, Dapper e SQL Server

  Quando está finalizando o DAPPER, e chega na parte de UPDATE a aula pula pra Execute Many em outra categoria, a de IMERSÃO. Verifiquei que quando começa aula de imersão ja está com o a função DeleteCategory finalizada mas não existe essa video aula, seguindo as mesmas ideias consegui fazer sozinho depois comparei com o código no GitHub.
  Mas pensei que seria necessário a exibição dessa aula para não gerar mais problemas futuros.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @gustavofusco, como vai?

Acredito que não gravei estas aulas pois acreditei ser irrelevante já que ambos métodos fazem uso do `Execute` já visto previamente.
Caso sinta-se confortável, coloca seu código aqui que comento se ver algo que possa ser melhorado.

**Comment by gusperandio** [comment]
Fiz o mesmo processo de antes, a diferença que coloquei um Guid por parametro

```csharp
static void DeleteCategory(SqlConnection conn, Guid IdDel)
{
    var deleteQuery = "Delete from Category where Id=@id";

    var rows = conn.Execute(deleteQuery, new{
        id = IdDel
    });

    string registro = rows > 1 ? "registros atualizados" : "registro atualizado";

    Console.WriteLine($"{rows} {registro}");
}
```

**Reply by andrebaltieri** [reply]
Me parece correto!!! É isto mesmo!
